### PR TITLE
Javadoc cleanup: model

### PR DESCRIPTION
**Pull request by sbesson**
Follow-up of https://github.com/openmicroscopy/openmicroscopy/pull/4031

As part of the build system cleanup, this PR reviews the Javadoc warnings for the model component of the server and removes unnecessary import headers.

To test this PR, check no more warning is reported for the `model` component when running `./build.py build-default`.

### CODE REVIEW

**Comment by jburel** [comment]
```<small> (<b>Internal version:</b> $Revision$ $Date$) </small>``` can also be removed since this is no longer set

**Comment by jburel** [comment]
I still have a large number of warning after running ```./build.py clean build-default```

```
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/util/SqlAction.java:108: warning - Tag @see:illegal character: "58" in "ticket:3961"
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/util/SqlAction.java:130: warning - @return tag has no arguments.
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/util/SqlAction.java:153: warning - @return tag has no arguments.
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/util/SqlAction.java:273: warning - @return tag has no arguments.
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/util/SqlAction.java:404: warning - @return tag has no arguments.
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/util/SqlAction.java:429: warning - @return tag has no arguments.
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/util/SqlAction.java:429: warning - Tag @see:illegal character: "58" in "ticket:3697"
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/util/SqlAction.java:429: warning - Tag @see:illegal character: "58" in "ticket:3253"
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/util/actions/PostgresSqlAction.java:68: warning - Tag @see: reference not found: ticket 3961
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/util/actions/PostgresSqlAction.java:68: warning - Tag @see: reference not found: ticket 9077
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/util/SqlAction.java:429: warning - Tag @see: reference not found: ticket:3697
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/util/SqlAction.java:429: warning - Tag @see: reference not found: ticket:3253
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/parameters/Filter.java:25: warning - @see tag has no arguments.
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/parameters/Filter.java:25: warning - Tag @see: reference not found: ome.parameters.Page
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/parameters/Filter.java:25: warning - Tag @see: reference not found: ome.parameters.Order
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/parameters/Filter.java:25: warning - Tag @see: reference not found: 
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/parameters/Parameters.java:49: warning - Tag @link: reference not found: IQuery
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/parameters/Parameters.java:49: warning - Tag @link: reference not found: IQuery#findByQuery(String, Parameters)
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/parameters/Parameters.java:49: warning - Tag @link: reference not found: IQuery#findAllByQuery(String, Parameters)
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/parameters/QueryParameter.java:31: warning - Tag @link: reference not found: ome.api.IQuery
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/util/MapPut.java:19: warning - End Delimiter } missing for possible See Tag in comment string: "Simple bean which calls "map.put(key, object)" on
 {@link InitializingBean#afterPropertiesSet(). Must adhere to the {@link Map}
 interface for nulls, etc."
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/util/PixelData.java:29: warning - Tag @see: reference not found: PixelBuffer
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/util/SqlAction.java:205: warning - Tag @link: missing '#': "findRepoDeleteLogs(DeleteLog)"
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/util/SqlAction.java:205: warning - Tag @link: missing '#': "deleteRepoDeleteLogs(DeleteLog)"
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/util/SqlAction.java:161: warning - Tag @link:illegal character: "60" in "#findRepoFile(String, String, String, Set<String>)"
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/util/SqlAction.java:161: warning - Tag @link:illegal character: "62" in "#findRepoFile(String, String, String, Set<String>)"
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/util/SqlAction.java:161: warning - Tag @link: can't find findRepoFile(String, String, String, Set<String>) in ome.util.SqlAction
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/util/SqlAction.java:179: warning - Tag @link: reference not found: OriginalFile
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/util/SqlAction.java:166: warning - Tag @link:illegal character: "60" in "#findRepoFile(String, String, String, Set<String>)"
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/util/SqlAction.java:166: warning - Tag @link:illegal character: "62" in "#findRepoFile(String, String, String, Set<String>)"
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/util/SqlAction.java:166: warning - Tag @link: can't find findRepoFile(String, String, String, Set<String>) in ome.util.SqlAction
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/util/SqlAction.java:108: warning - Tag @see: reference not found: ticket:3961
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/util/SqlAction.java:179: warning - Tag @link: reference not found: OriginalFile
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/util/SqlAction.java:179: warning - @param argument "mimetype" is not a parameter name.
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/util/SqlAction.java:179: warning - Tag @link: reference not found: OriginalFile
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/util/SqlAction.java:197: warning - @param argument "uuid" is not a parameter name.
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/util/SqlAction.java:429: warning - Tag @see: reference not found: ticket:3697
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/util/SqlAction.java:429: warning - Tag @see: reference not found: ticket:3253
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/util/SqlAction.java:463: warning - @param argument "group" is not a parameter name.
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/util/SqlAction.java:179: warning - Tag @link: reference not found: OriginalFile
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/util/SqlAction.java:108: warning - Tag @see: reference not found: ticket:3961
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/util/SqlAction.java:179: warning - Tag @link: reference not found: OriginalFile
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/util/SqlAction.java:179: warning - Tag @link: reference not found: OriginalFile
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/util/TableIdGenerator.java:29: warning - Tag @link: reference not found: OptimizerFactory.PooledOptimizer
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/util/Utils.java:104: warning - @return tag has no arguments.
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/util/Utils.java:33: warning - @DEV.TODO is an unknown tag.
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/model/IAnnotationLink.java:16: warning - Tag @link: reference not found: Annotation
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/model/IGlobal.java:17: warning - Tag @link: reference not found: Details
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/model/IGlobal.java:17: warning - Tag @link: reference not found: Details
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/model/ILink.java:30: warning - Tag @see: reference not found: ome.model.containers.Category
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/model/ILink.java:30: warning - Tag @see: reference not found: ome.model.containers.CategoryGroup
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/conditions/ExpiredCredentialException.java:16: warning - Tag @see: reference not found: ome.api.ISession#changeExpiredCredentials(String, String, String)
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/conditions/ReadOnlyGroupSecurityViolation.java:19: warning - Tag @see:illegal character: "58" in "ticket:1769"
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/conditions/ReadOnlyGroupSecurityViolation.java:19: warning - Tag @see:illegal character: "58" in "ticket:1992"
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/conditions/ReadOnlyGroupSecurityViolation.java:19: warning - Tag @see: reference not found: ticket:1769
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/conditions/ReadOnlyGroupSecurityViolation.java:19: warning - Tag @see: reference not found: ticket:1992
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/conditions/RemovedSessionException.java:16: warning - Tag @link: reference not found: Session
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/conditions/SessionException.java:19: warning - Tag @link: reference not found: ISession
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/conditions/SessionException.java:19: warning - Tag @link: reference not found: ome.api.ISession
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/conditions/ValidationException.java:31: warning - Tag @link: reference not found: ome.model.containers.Category Categories
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/conditions/ValidationException.java:31: warning - Tag @link: reference not found: ome.model.containers.CategoryGroup CategoryGroup
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/model/internal/Details.java:42: warning - Tag @link: reference not found: ome.api.IUpdate 
 update
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/model/internal/Details.java:42: warning - Tag @see: reference not found: ome.api.IUpdate
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/model/internal/GraphHolder.java:53: warning - Tag @link: reference not found: ome.api.IUpdate
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/model/internal/GraphHolder.java:53: warning - Tag @link: reference not found: ome.api.IUpdate
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/model/internal/Permissions.java:213: warning - Tag @link: reference not found: Permission
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/model/internal/Permissions.java:213: warning - Tag @see: reference not found: ome.tools.hibernate.SecurityFilter
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/model/internal/Permissions.java:224: warning - Tag @see: reference not found: ome.tools.hibernate.SecurityFilter
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/model/units/BigResult.java:36: warning - Tag @link: reference not found: double
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/model/units/BigResult.java:36: warning - Tag @link: reference not found: double
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/model/units/Conversion.java:199: warning - Tag @link: can't find pow(BigDecimal) in java.math.BigDecimal
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/conditions/RemovedSessionException.java:16: warning - Tag @link: reference not found: Session
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/model/IAnnotationLink.java:16: warning - Tag @link: reference not found: Annotation
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/model/IGlobal.java:17: warning - Tag @link: reference not found: Details
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/parameters/QueryParameter.java:31: warning - Tag @link: reference not found: ome.api.IQuery
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/parameters/package.html: warning - Tag @see: reference not found: ome.api.IQuery
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/parameters/Filter.java:31: warning - Tag @link: reference not found: ome.services.query.Query
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/conditions/RemovedSessionException.java:16: warning - Tag @link: reference not found: Session
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/conditions/RemovedSessionException.java:16: warning - Tag @link: reference not found: Session
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/model/IAnnotationLink.java:16: warning - Tag @link: reference not found: Annotation
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/model/IAnnotationLink.java:16: warning - Tag @link: reference not found: Annotation
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/model/IGlobal.java:17: warning - Tag @link: reference not found: Details
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/model/internal/GraphHolder.java:53: warning - Tag @link: reference not found: ome.api.IUpdate
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/model/IAnnotationLink.java:16: warning - Tag @link: reference not found: Annotation
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/model/IGlobal.java:17: warning - Tag @link: reference not found: Details
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/model/IAnnotationLink.java:16: warning - Tag @link: reference not found: Annotation
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/model/IGlobal.java:17: warning - Tag @link: reference not found: Details
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/model/IGlobal.java:17: warning - Tag @link: reference not found: Details
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/model/IGlobal.java:17: warning - Tag @link: reference not found: Details
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/parameters/QueryParameter.java:31: warning - Tag @link: reference not found: ome.api.IQuery
Building index for all the packages and classes...
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/util/SqlAction.java:179: warning - Tag @link: reference not found: OriginalFile
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/model/IAnnotationLink.java:16: warning - Tag @link: reference not found: Annotation
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/model/IGlobal.java:17: warning - Tag @link: reference not found: Details
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/parameters/QueryParameter.java:31: warning - Tag @link: reference not found: ome.api.IQuery
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/conditions/RemovedSessionException.java:16: warning - Tag @link: reference not found: Session
/Users/jmarie/Documents/git-repo/openmicroscopy/components/model/src/ome/model/internal/GraphHolder.java:53: warning - Tag @link: reference not found: ome.api.IUpdate
```

**Comment by sbesson** [comment]
Can you paste your SHA1? Most of these warnings look exactly like the ones fixed as part of this PR. Furthermore, from https://ci.openmicroscopy.org/view/DEV/job/OMERO-DEV-merge-build/ICE=3.5,jdk=8_LATEST,label=octopus/58/consoleText, I have

```

package-extra:
Building jar: /opt/hudson/workspace/OMERO-DEV-merge-build/src/components/model/target/model-psql-sources.jar
Created dir: /opt/hudson/workspace/OMERO-DEV-merge-build/src/components/model/target/javadocs
Generating Javadoc
Javadoc execution
Loading source file /opt/hudson/workspace/OMERO-DEV-merge-build/src/components/model/src/ome/conditions/ApiUsageException.java...
...
Loading source file /opt/hudson/workspace/OMERO-DEV-merge-build/src/components/model/target/generated/src/ome/model/stats/StatsInfo.java...
Constructing Javadoc information...
Standard Doclet version 1.8.0_51
Building tree for all the packages and classes...
Building index for all the packages and classes...
Building index for all classes...
Building jar: /opt/hudson/workspace/OMERO-DEV-merge-build/src/components/model/target/model-psql-javadoc.jar

```

**Comment by jburel** [comment]
sha1:  b5120c7a6c60a3514adadde1cef14dd8cf7ed7d6

**Comment by sbesson** [comment]
Strange. And your repository is clean? Going through your warnings list, I do not see any that was not fixed in 9b36c9f.

**Comment by jburel** [comment]
I re-ran again and this time it is fine..
ignore previous report.

**Comment by sbesson** [comment]
@jburel: pushed commits cleaning  `Internal version`

**Comment by jburel** [comment]
Changes look good.

**Comment by jburel** [comment]
Will check wiht @sbesson when he returns, to double-check if he has more commits.

**Comment by sbesson** [comment]
@jburel: no more commit planned on my side for this PR.

File: components/model/src/ome/model/internal/Permissions.java [hunk 1]
```diff
@@ -205,10 +205,8 @@ public boolean isSet(Flag flag) {
 
     /**
      * returns the order of the bit representing the given {@link Flag}. This
-     * is dependent on the internal representation of {@link Permission} and
+     * is dependent on the internal representation of {@link Permissions} and
      * should only be used when necessary.
-     *
-     * @see ome.tools.hibernate.SecurityFilter
```
Comment (joshmoore) [inline components/model/src/ome/model/internal/Permissions.java]: Note: losing some of these is a bit sad. It definitely points to something circular (or at least forward referencing) but sometimes that will be necessary.

Comment (sbesson) [inline components/model/src/ome/model/internal/Permissions.java]: Agreed, but we need to use another semantics than

```
@see package.class
```

due to the forward referencing. Not sure what the correct way is for Javadocs.

Comment (joshmoore) [inline components/model/src/ome/model/internal/Permissions.java]: http://docs.oracle.com/javase/7/docs/technotes/tools/windows/javadoc.html#see

```
@see "string"
```

would be ok.

Comment (sbesson) [inline components/model/src/ome/model/internal/Permissions.java]: Makes sense. Do you have a clear list of the ones you want to keep? Want to create a follow-up item to restore them https://trello.com/c/RpJxX7S7/35-javadoc-cleanup?

Comment (joshmoore) [inline components/model/src/ome/model/internal/Permissions.java]: In general I'd keep all forward references to our own classes. In this case, that's likely things to `ome.security` or `ome.tools`. (Which points to these potentially wanting to be decoupled :smile:)

Comment (sbesson) [inline components/model/src/ome/model/internal/Permissions.java]: See https://github.com/openmicroscopy/openmicroscopy/pull/4096

**Pull request merged**